adapters/repos/db/index.go

```diff
@@ -1228,11 +1228,12 @@ func (i *Index) isLocalShard(shard string) bool {
 }
 
 func (i *Index) mergeObject(ctx context.Context, merge objects.MergeDocument,
-	replProps *additional.ReplicationProperties,
+	replProps *additional.ReplicationProperties, tenantKey string,
 ) error {
 	i.backupStateLock.RLock()
 	defer i.backupStateLock.RUnlock()
-	shardName, err := i.shardFromUUID(merge.ID)
+
```

Comment:
I think the validation below is missing at this point:
```
	if err := i.validateMultiTenancy(tenantKey); err != nil {
		return err
	}
```

Same fix applies to `deleteObject()`

Reply:
You are right, fixed.